protocol/deposits-interop.md

```diff
@@ -0,0 +1,189 @@
+analysis of transactions. By only allowing for EOAs to send cross chain transactions, this meant that the
+calldata for the `CrossL2Inbox` would be present in the transaction's data itself. This could be statically
+analyzed by both the protocol (fork choice rule) and block builders. Forcing EOAs to send the executing
+messages was the biggest complaint by people as it greatly reduced the design space, prevented compatibility
+with smart contract wallets and prevented compatibility with [ERC-4337](https://eips.ethereum.org/EIPS/eip-4337) bundlers.
+
+With the removal of the "only EOA invariant", execution is now a requirement to determine the validity of a transaction.
+This applies for deposit transactions as well. The derivation pipeline needs some sort of legibility into the validity
+of these deposit transactions.
+
+To prevent race conditions on deposited executing messages that reference recent initiating messages, a delay should
+be enforced on deposits that contain executing messages. This prevents attacks where some amount of p2p latency
+can cause nodes to inconsistent chains.
+
+# Proposed Solution
+
+The stateless injective function for mapping deposit events into deposit transactions is updated to be a
+2-valued function, meaning that any input can be mapped to one of two inputs. In mathematical terms:
+
+$$f:A→\{b1,b2\}$$
+
+A deposit transaction event can be mapped to either a valid deposit transaction or a noop deposit transaction.
+
+A new invariant called the "sequencing window invariant" is introduced that enforces a delay for deposit
+transactions that trigger executing messages.
+
+## Derivation Pipeline Execution
+
+The derivation pipeline is updated to include EVM execution of the deposit transactions. A forking EVM provider backed
+by L2 tip state is used. This enables simulation of the deposit transactions before they are included in L2. If the deposit
+transaction fails the sequencing window invariant or creates an invalid executing message, it is mapped into a noop deposit
+transaction.
+
+A noop deposit transaction:
+- keeps its sourcehash for tx hash uniqueness
+- keeps its from, so the nonce can increment
+- keeps its to, in case its also depositing `ether`
+- keeps its mint, in case its depositing `ether`
+- gas is set to 0
+- data is set to empty
+
+This should hold the invariant that the `ether` is always minted when sent into the `OptimismPortal` but prevents
+any additional execution on top of that.
+
+In pseudocode:
+
+```python
+txs = []
+evm = prepare_evm(eth)
+for i, tx in enumerate(payload.transactions):
+  if tx.type() != DEPOSIT_TYPE:
+    tx.append(tx)
+    continue
+  logs = evm.call(tx.to_message())
+  for log in logs:
+    if is_executing_message(log):
+      if sequencing_window_invariant(log) == False or is_valid_message(log) == False:
+        tx.gas = 0
+        tx.data = ""
+        txs.append(tx)
+        break
+```
+
```

Comment:
This logic can be added to the derivation pipeline pretty easily [here](https://github.com/ethereum-optimism/optimism/blob/93891de2176f6f997da80a4de8de84466164bb25/op-node/rollup/derive/attributes.go#L70)